### PR TITLE
Remove skipping of certain tests

### LINKED ISSUE #1419: Remove skipping of certain tests
Due to an update of devnet [#547](https://github.com/0xSpaceShard/starknet-devnet-rs/pull/547) the bug was removed which now affects our tests. Declaring multiple contracts with the same class hash isn't possible right now.
The short-term solution was to skip these tests.

### PR DESCRIPTION
**Pull request by kkawula**
Closes #1419

In the sense of the solution, duplicates of contracts that have to be redeclared was added, because contracts are compiled before tests, we cannot salt them in runtime. Additionally, these duplicates were added only in cairoV2 version, to avoid massive code duplication. However, these tests are only used for docs purposes, so this functionality is still covered by other tests that involve two versions of Cairo.

## Introduced changes

- Added copy of contracts that have to be declared more than one time

##

- [ ] This PR contains breaking changes

### CODE REVIEW

File: starknet_py/tests/e2e/block_test.py
Comment (franciszekjob) [inline starknet_py/tests/e2e/block_test.py:1]: Looks like [this line](https://github.com/software-mansion/starknet.py/pull/1470/files#diff-129e49f80171fec9a84a1e967524c2b0c039dde352ad130be0736198f6a6d83aL67) should be removed too?

File: starknet_py/tests/e2e/docs/code_examples/test_contract.py [hunk 1]
```diff
@@ -66,7 +64,7 @@ async def test_declare_v2(account):
         account,
         compiled_contract=compiled_contract["sierra"],
         compiled_contract_casm=compiled_contract["casm"],
-        max_fee=int(1e15),
+        max_fee=int(1e18),
```
Comment (franciszekjob) [inline starknet_py/tests/e2e/docs/code_examples/test_contract.py:67]: What's the reason for this change? Btw, we can use `MAX_FEE` constant.

File: starknet_py/tests/e2e/docs/guide/test_declaring_contracts.py [hunk 2]
```diff
@@ -18,7 +22,7 @@ async def test_declaring_contracts(account, map_compiled_contract_and_class_hash
         compiled_contract=compiled_contract,
         compiled_class_hash=class_hash,
         l1_resource_bounds=ResourceBounds(
-            max_amount=5000, max_price_per_unit=int(1e12)
+            max_amount=int(1e5), max_price_per_unit=int(1e13)
```
Comment (franciszekjob) [inline starknet_py/tests/e2e/docs/guide/test_declaring_contracts.py:25]: Could you please explain why did we change this?

File: starknet_py/tests/e2e/docs/guide/test_declaring_contracts.py [hunk 3]
```diff
@@ -1,13 +1,17 @@
...
-# TODO (#1419): Fix contract redeclaration
-@pytest.mark.skip(reason="Redeclaration occurred")
+@pytest.mark.skipif(
+    "--contract_dir=v2" not in sys.argv,
+    reason="Contract exists only in v2 directory",
+)
 @pytest.mark.asyncio
-async def test_declaring_contracts(account, map_compiled_contract_and_class_hash):
-    (compiled_contract, class_hash) = map_compiled_contract_and_class_hash
+async def test_declaring_contracts(account, map_compiled_contract_and_class_hash_copy1):
```
Comment (franciszekjob) [inline starknet_py/tests/e2e/docs/guide/test_declaring_contracts.py:13]: nit: Also please apply it to other variables 🙏 .
```suggestion
async def test_declaring_contracts(account, map_compiled_contract_and_class_hash_copy_1):
```

File: starknet_py/tests/e2e/docs/code_examples/test_contract.py (hunk 1 above)
Comment (kkawula) [reply starknet_py/tests/e2e/docs/code_examples/test_contract.py:67]: The cost of declaration changed since this contract started being skipped, so it was failing due to `Max fee is smaller than the minimal transaction cost` error .
The content of this test code appears in the documentation, so I suppose that using `MAX_FEE` could be confusing

Comment (franciszekjob) [reply starknet_py/tests/e2e/docs/code_examples/test_contract.py:67]: Ok, I didn't see it's in docs, thanks for explanation.

**Review by franciszekjob — APPROVED** [review]
LGTM
